.circleci/config.yml

```diff
@@ -32,22 +36,77 @@ jobs:
     steps:
       - checkout
       - restore_cache:
-          keys:
-            - node-v1-{{ .Branch }}-{{ checksum "package-lock.json" }}
-            - node-v1-{{ .Branch }}-
-            - node-v1-
-      - run: npm install --no-save
+          key: frontend-v1-{{ checksum "package-lock.json" }}
+      # Only install if node_modules wasn’t cached.
+      - run: |
+            if [[ ! -e "node_modules" ]]; then
+                npm install --no-save --no-optional --no-audit --no-fund --progress=false
```

Comment:
Is there a reason you don't use `npm ci` here ?

Reply:
Last time I tried it, it was slower than this command, which does the same thing anyway when there isn’t a `node_modules` folder. I’ll try again and see how they compare these days.

Reply:
I think - it might have changed though - that `npm ci` throws an error if one of the dependencies is missing from the `package-lock.json` file. 

I switched to `npm ci` in a project - a while back - because a minor version of an npm package was removed from the registry and the CI installed the next version, causing difference between local package version and the one installed in the CI, introducing bugs on production that were not present locally. 
However, it was also caused by badly fixed version in the `package.json`...

Reply:
https://docs.npmjs.com/cli/v8/commands/npm-ci
> If dependencies in the package lock do not match those in package.json, npm ci will exit with an error, instead of updating the package lock.

Well, it seems it is still the case. However better speed might be more important in that case than covering a situation that would almost never happen.

Reply:
From quick timing, `npm ci` was 1s faster. Time to switch 😄